provisioning/roles/docker/tasks/main.yml

```diff
@@ -1,10 +1,71 @@
-- name: Install Docker
+- name: Add docker repository
+  yum_repository:
+    name: docker-ce-stable
+    state: present
+    description: Docker CE Stable - $basearch
+    baseurl: "https://download.docker.com/linux/centos/7/$basearch/stable"
+    enabled: 1
+    gpgcheck: 1
+    repo_gpgcheck: 1
+    gpgkey:
+      - "https://download.docker.com/linux/centos/gpg"
+
+- name: Install Docker prerequisites
   yum:
-    name: docker
+    name: "{{ item }}"
     state: latest
+  with_items:
+    - device-mapper-persistent-data
+    - lvm2
+
+- name: Install Docker
+  # Note: as of writing, latest supported Docker version by Kubernetes is 17.03
+  # Note: we use a command instead of the yum module since we need to set additional options not supported by Ansible
+  command: yum install -y --setopt=obsoletes=0 docker-ce-17.03.2.ce-1.el7.centos.x86_64
 
 - name: Enable and start Docker daemon
   systemd:
     name: docker
     enabled: yes
     state: started
+
+- name: Add nvidia-docker repositories
+  yum_repository:
+    name: "{{ item }}"
+    state: present
+    description: "{{ item }}"
+    baseurl: "https://nvidia.github.io/{{ item }}/centos7/$basearch"
+    enabled: 1
+    gpgcheck: 0
+    repo_gpgcheck: 1
+    gpgkey:
+      - "https://nvidia.github.io/{{ item }}/gpgkey"
+  with_items:
+    - libnvidia-container
+    - nvidia-container-runtime
+    - nvidia-docker
+
+- name: Install nvidia-docker
+  yum:
+    name: "{{ item }}"
+    state: latest
+  with_items:
+    - nvidia-container-runtime-2.0.0-1.docker17.03.2.x86_64
+    - nvidia-docker2-2.0.3-1.docker17.03.2.ce.noarch
+
+- name: Fix cgroup drivers in Docker daemon

```

Comment:
Never overwrite the package installed service file, copy the orginal to `/etc/systemd/system/docker.service` and update that file instead. It is also possible to create a "patch" file for system services which only defines the `ExecStart`, which would mean you don't need to update and only need to upload a service file and `systemctl daemon-reload` (aka Drop-in files, see https://wiki.archlinux.org/index.php/systemd#Editing_provided_units ).